mhyeun/twitter-emojify-api
# Create Flask endpoint at /emojify-tweets

### PR Summary

This PR creates a Flask GET endpoint at `/emojify-tweets` which accepts mandatory query params `username` and `tweets`. It will then return a `JSON` list of emojified tweets. It will return a `400` error if the params are not given or if the username is invalid (not found).

This PR also creates a `/health` route which just returns a `200`.

Screenshot of working example:

![image](https://user-images.githubusercontent.com/46268420/103609747-883ed680-4eec-11eb-8a5f-3bda1301ebeb.png)

![image](https://user-images.githubusercontent.com/46268420/103609387-bf60b800-4eeb-11eb-9c45-9accafcc89ed.png)

#### How to test this endpoint 🧪 
To run the Flask server, run:

```sh
export FLASK_APP=src/emojify_tweets.py
flask run
```

The server should be running on port 5000, i.e. `localhost:5000`.

To test the endpoint, make a GET request to `localhost:5000/emojify-tweets` with param `username` being the Twitter username and `tweets` being the number of tweets to emojify. For example, to make a request to emojify 5 of AOC's tweets, make a request to:
```
localhost:5000/emojify-tweets?username=aoc&tweets=5
```

_Not working?_
Try making a request to `/health` to see if the server is even up.
